pallets/otc-settlements/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+
+		ensure!(
+			asset_a_balance_after == asset_a_balance_before,
+			Error::<T>::BalanceInconsistency
+		);
+		ensure!(
+			asset_b_balance_after == asset_b_balance_before,
+			Error::<T>::BalanceInconsistency
+		);
+
+		Self::deposit_event(Event::Executed {
+			asset_id: asset_a,
+			profit,
+		});
+
+		Ok(())
+	}
+
+	/// Store the latest block number in the offchain storage.
+	/// Returns `true` if `block_number` is newer than the block number stored in the storage.
+	fn try_update_last_block_storage(block_number: BlockNumberFor<T>) -> bool {
+		let last_update_storage = StorageValueRef::persistent(OFFCHAIN_WORKER_DATA_LAST_UPDATE);
+		let last_update = last_update_storage
+			.get::<BlockNumberFor<T>>()
+			.unwrap_or_default()
+			.unwrap_or_default();
+
+		if block_number > last_update {
+			last_update_storage.set(&block_number);
+			true
+		} else {
+			false
+		}
+	}
+
+	/// Sort open OTCs orders and save a list in the offchain storage.
+	fn sort_otcs(block_number: BlockNumberFor<T>) {
+		log::debug!(
+			target: "offchain_worker::sort_otcs",
+			"sort_otcs()");
+
+		// acquire offchain worker lock.
+		let lock_expiration = Duration::from_millis(LOCK_TIMEOUT_EXPIRATION);
+		let mut lock = StorageLock::<'_, Time>::with_deadline(SORTED_ORDERS_LOCK, lock_expiration);
+
+		if Self::try_update_last_block_storage(block_number) {
+			if let Ok(_guard) = lock.try_lock() {
+				let sorted_otcs = StorageValueRef::persistent(OFFCHAIN_WORKER_DATA);
+
+				let mut list = vec![];
+				for (otc_id, otc) in <pallet_otc::Orders<T>>::iter() {
+					let otc_price = FixedU128::checked_from_rational(otc.amount_out, otc.amount_in);
+
+					let route = T::Router::get_route(AssetPair {
+						asset_in: otc.asset_out,
+						asset_out: otc.asset_in,
+					});
+					let router_price_before = T::Router::spot_price_with_fee(&route.clone());
+
+					if let (Some(otc_price), Some(router_price)) = (otc_price, router_price_before) {
+						// otc's with no arb opportunity are at the end of the list and are not sorted
+						let price_diff = otc_price.saturating_sub(router_price);
+						list.push((otc_id, otc_price, router_price, price_diff));
+					}
+				}
+
+				// sort the list by the price diff
+				list.sort_by(|a, b| b.3.partial_cmp(&a.3).unwrap());
+
+				sorted_otcs.set(&list);
+			};
+		}
+	}
+
+	/// Iterate over sorted list of OTCs and try to find arbitrage opportunities.
+	fn settle_otcs() {
+		log::debug!(
+			target: "offchain_worker::settle_otcs",
+			"settle OTCs");
+		// iterate over sorted OTCs
+		let sorted_otcs_storage = StorageValueRef::persistent(OFFCHAIN_WORKER_DATA);
+		let sorted_otcs = sorted_otcs_storage
+			.get::<Vec<SortedOtcsStorageType>>()
+			.unwrap_or_default()
+			.unwrap_or_default();
+
+		for (otc_id, otc_price, router_price_before, _price_diff) in sorted_otcs.iter() {
+			log::debug!(
+			target: "offchain_worker::settle_otcs",
+				"test OTC id {:?} ", otc_id);
+
+			if router_price_before > otc_price {
```

Comment:
just curious - if you ignoring this case, why put it in the list at all ? 

on line 426, where you add it to this list-  you can already remove this one.

Reply:
Thanks. Updated.